### PR TITLE
clang-format: Handle Objective-C(++) in config

### PR DESCRIPTION
**Pull request by fwcd**
Since there are now a couple of PRs adding Objective-C++ code, mainly for interfacing with Apple frameworks on macOS (e.g. #11353, #4809), we need to make sure that clang-format is configured correctly for this. While the aforementioned PRs bundle this patch too, I've extracted it to this PR for easier reviewability.

While clang-format has no trouble handling Objective-C++, the temporary line-breaking config currently only works on C++ (since it is based on the config dumped for a `.cpp` file). For this reason, this PR adds a second mini-configuration for `Language: ObjC` that uses a column limit of 80 and excludes Objective-C(++) from the second pass.

### CODE REVIEW

**Comment by Swiftb0y** [comment]
Why are we doing this in the clang-format script? Couldn't we just add a fixed column-limit to our base clang-format config?

**Comment by fwcd** [comment]
The issue is that this temporary config is generated from a `.cpp` file, therefore it only includes C++ rules and causes an exception when trying to format ObjC. The base clang-format has no issues with ObjC.

As I've suggested, once we have some ObjC in the codebase, we can do this in a cleaner way by basing the temporary config on the config for an actual `.m` or `.mm` file, but we can't do that yet.

**Comment by fwcd** [comment]
But I agree, the current solution already feels a bit like a workaround, so perhaps we'd want to get rid of the 'dual-pass' clang-format if we decide to clean up here anyway...

**Comment by Swiftb0y** [comment]
Well IIUC the first pass uses the unmodified clang-format in the repo root and the second pass uses the temporary one, right? So can't we just exclude `.m` and `.mm` files from that pass?

**Comment by fwcd** [comment]
I've updated the second pass to operate only on `.cpp` files (which should make sense given that the temporaryclang-format config is based on a dumped C++-only config) and added a minimal `Language: ObjC` config to the main `.clang-format` that wraps lines to 80 characters. This is slightly more restrictive than the C++ rule, which allows lines up to 100 chars but wraps at 80 if exceeded, but since we probably won't include much Objective-C(++) anyway, this hopefully shouldn't be much of an issue.

**Comment by daschuer** [comment]
With the current state ObjC files are formatted a different style compared to the rest of Mixxx. That makes it less smooth switching form one file to another. I would prefer to either disable the formatting or duplicate the Cpp solution. 

The main reason for the two passes is a workaround for the issue of: https://github.com/llvm/llvm-project/issues/58417
It which keeps manual line breaks in place. This means it does not un-breaks manual line breaks. 
The fuzzy 100/80 column limit is only a nice side effect.

**Comment by fwcd** [comment]
Perhaps we could replace the hard 100/80 rule with [penalties](https://stackoverflow.com/questions/26635370/in-clang-format-what-do-the-penalties-do) and remove the second pass altogether?

**Comment by daschuer** [comment]
Since we will only have a view ObjC files the fixed column limit would work for me (to release you from the work).
If you just add the relevant definition to the ObjC region, it is ready to go.

File: tools/clang_format.py [hunk 1]
```diff
@@ -28,8 +28,9 @@ def get_clang_format_config_with_columnlimit(rootdir, limit):
         text=True,
     )
     proc.check_returncode()
+
     return re.sub(
-        r"(ColumnLimit:\s*)\d+", r"\g<1>{}".format(BREAK_BEFORE), proc.stdout
+        r"(ColumnLimit:\s*)\d+", r"\g<1>{}".format(limit), proc.stdout
```
Comment (fwcd) [inline tools/clang_format.py:33]: \+ this minor fix, the function didn't use its `limit` argument before and used the constant directly (though with the same effect)

**Review by fwcd — COMMENTED** [review]
This should already be the current state of the branch

**Comment by daschuer** [comment]
> This should already be the current state of the branch

Most settings are still in the Cpp only region and are not applied to ObjC.

**Comment by fwcd** [comment]
Ah I see what you mean, I've added the Cpp rules to the ObjC section. This unfortunately causes some redundancy there, but gets us consistent formatting across C++ and Objective-C(++).

**Review by Swiftb0y — APPROVED** [review]
LGTM

**Comment by fwcd** [comment]
I've rebased it to 2.4

**Review by Swiftb0y — APPROVED** [review]

**Comment by Swiftb0y** [comment]
@daschuer merge?

**Review by daschuer — APPROVED** [review]
LGTM, Thank you

**Pull request merged**